File: client/src/app/model/vehicle-info-request.ts
import {TyreCondition} from "./enums/tyre-condition";

export class VehicleInfoRequest {

  vehicle_id: number;
  is_maintenance_required?: boolean;
  is_clean?: boolean;
  vehicle_usage_time_in_months?: number;
  tired_condition?: TyreCondition;

  constructor(
    vehicle_id: number,
    is_maintenance_required?: boolean,
    is_clean?: boolean,
    vehicle_usage_time_in_months?: number,
    tired_condition?: TyreCondition,
  ) {
    this.vehicle_id = vehicle_id || 0;
    this.is_maintenance_required = is_maintenance_required !== undefined ? is_maintenance_required : undefined;
    this.is_clean = is_clean !== undefined ? is_clean : undefined;
    this.vehicle_usage_time_in_months = vehicle_usage_time_in_months || 0;
    this.tired_condition = tired_condition || TyreCondition.FAIR;
  }
}

File: src/main/java/com/project/vrs/resources/request/VehicleInfoRequest.java
package com.project.vrs.resources.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.vrs.enums.TyreCondition;
import jakarta.validation.constraints.Min;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VehicleInfoRequest {

    @JsonProperty("vehicle_id")
    private Long vehicleId;

    @JsonProperty("is_maintenance_required")
    private boolean isMaintenanceRequired;

    @JsonProperty("is_clean")
    private boolean isClean;

    @JsonProperty("is_available")
    private Boolean isAvailable;

    @JsonProperty("vehicle_usage_time_in_months")
    @Min(value=0 ,message ="Minimum value must be 0.")
    private int vehicleUsageTimeInMonths;

    @JsonProperty("tired_condition")
    private TyreCondition tireCondition;
}
